### PR TITLE
Add Candidate Tests page

### PR DESCRIPTION
**Pull request by howard-e**
Adds the Candidate Tests list page based on the latest mockups described in #437. 

To view the page, log in as an admin and navigate to `localhost:<PORT>/candidate-tests`. Please also ensure you've ran the latest migrations.

```sh
yarn sequelize:test db:migrate
yarn db-import-tests:dev;
```

### CODE REVIEW

**Review by evmiguel — COMMENTED** [review]
The page looks awesome! One thing I noticed is, when I mark a Test Plan as Draft, then mark it as In Review - the Candidate Tests Page doesn't refresh to show the plan again.

File: client/components/TestQueue/queries.js [hunk 1]
```diff
@@ -323,6 +323,23 @@ export const UPDATE_TEST_PLAN_REPORT_MUTATION = gql`
     }
 `;
 
+export const UPDATE_TEST_PLAN_REPORT_RECOMMENDED_TARGET_DATE_MUTATION = gql`
+    mutation UpdateTestPlanReportStatus(
```
[the remark this answers is not in the document]
Comment (howard-e) [reply client/components/TestQueue/queries.js:327]: Good catch, thanks!

File: client/components/CandidateTests/TestPlans/index.jsx [hunk 2]
```diff
@@ -0,0 +1,1067 @@
...
+                                                                <Dropdown.Item
+                                                                    role="menuitem"
+                                                                    onClick={async () => {
+                                                                        await updateReportStatus(
+                                                                            testPlanReports,
+                                                                            'FINALIZED'
+                                                                        );
+                                                                    }}
+                                                                    disabled={
+                                                                        true /*TODO: Enabled only if all existing TestPlanReports across browsers have been accepted*/
```
[the remark this answers is not in the document]
Comment (howard-e) [reply client/components/CandidateTests/TestPlans/index.jsx:688]: Nope, my own TODO that I forgot to address! Will do so now, thanks for catching that.

On that note, the only other major outstanding implementation that comes to mind here is presenting the vendor representative only with their assigned AT after syncing up the changes with #461 to make use of the `vendorCompany` attribute.

**Comment by howard-e** [comment]
> One thing I noticed is, when I mark a Test Plan as Draft, then mark it as In Review - the Candidate Tests Page doesn't refresh to show the plan again.

@evmiguel thanks for a taking a look. After briefly looking, it seems to affect the `/reports` endpoint as well. Will take a look at why.

**Review by evmiguel — APPROVED** [review]

**Pull request merged**